databases/week1/homework.sql

```diff
@@ -0,0 +1,48 @@
+-- Find out how many tasks are in the task table
+SELECT COUNT(id)
+FROM task;
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id)
+FROM task
+WHERE due_date IS NULL;
+-- Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id=(SELECT id FROM status WHERE name='done'
+ );
+ -- Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE status_id IN (SELECT id FROM status WHERE name !='done'
+ );
+ -- Get all the tasks, sorted with the most recently created first
+ SELECT *
+ FROM task
+ ORDER BY created DESC;
+ -- Get the single most recently created task
+ SELECT *
+ FROM task
+ ORDER BY created DESC
+ LIMIT 1;
+ -- Get the title and due date of all tasks where the title or description contains database
+ SELECT title, due_date
+ FROM task
+ WHERE title LIKE '%database%' OR description LIKE '%database%';
+ -- Get the title and status (as text) of all tasks
+ SELECT title, name
+ FROM task t
+ JOIN status s
+      ON t.status_id=s.id;
+-- Get the name of each status, along with a count of how many tasks have that status
+ SELECT name, COUNT(s.id) AS 'count'
+ FROM task t
+ JOIN status s
+      ON t.status_id=s.id
+ GROUP BY s.id;
```

Comment:
The `group by` should use `s.name`. (we are going to explain it again in class)
and the count should use tasks `id`

```suggestion
 SELECT s.name, COUNT(t.id) AS 'count'
 FROM task t
 JOIN status s
      ON t.status_id=s.id
 GROUP BY s.name;
```